services/src/main/java/org/keycloak/privacy/PrivacyProvider.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright 2020 Red Hat, Inc. and/or its affiliates
+ * and other contributors as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.keycloak.privacy;
+
+import org.keycloak.events.Event;
+import org.keycloak.provider.Provider;
+
+/**
+ * Allows to filter input values that might contain sensitive personally identifiable information (PII).
+ *
+ * @author <a href="mailto:thomas.darimont@googlemail.com">Thomas Darimont</a>
+ */
+public interface PrivacyProvider extends Provider {
```

Comment:
Wonder if we should add "String filter (String input)", then have a default method for the current two methods in here that just uses that. That would make it simpler to implement one if you don't care about type, etc. and also means you can have a fallback if there's additional things added in the future.